### DISCUSSION TITLE
How do I use mutationCache

### DISCUSSION THREAD

**Discussion by ojknation** [TanStack/query#2252] [Q&A]
Hello, please I need help on how to use `mutationCache` to retreive previous mutation data from the cache, is this a valid use case for it? if yes how do I go about it, I initially tried using it's `getAll()` method within a component  but I keep getting an empty array. I later saw in the docs that it's an option under the **QueryClient**  but I don't know what should go in there to enable the `mutationCache` work.

![image](https://user-images.githubusercontent.com/37719470/117520297-67bc8100-af9f-11eb-8bc3-b9fbd659addc.png)
![image](https://user-images.githubusercontent.com/37719470/117520301-6b500800-af9f-11eb-927a-53d1b16a8ad1.png)

Thank you.

**Comment by TkDodo** [comment]
What do you need it for?

**Reply by spacepolice10** [reply]
Hey, 

It would be quite useful to work with like with simple local state management in some tasks like file uploading. Connecting a real local state manager seems to be overkill in such situations and search parameters from route not really fit there because of strong connection with the server

**Reply by TkDodo** [reply]
react query is not a local state manager.
